pkg/provider/helm/helm.go

```diff
@@ -55,10 +55,31 @@ type HelmSPIREProvider struct {
 	cluster          *clusterpb.Cluster
 }
 
-func NewHelmSPIREProvider(ctx context.Context, cluster *clusterpb.Cluster, spireValues, spireCRDsValues map[string]any) (*HelmSPIREProvider, error) {
+type providerOptions struct {
+	kubeConfigPath string
+}
+
+type Option func(*providerOptions)
+
+func WithKubeConfig(path string) Option {
+	return func(opts *providerOptions) {
+		opts.kubeConfigPath = path
+	}
+}
+
+func NewHelmSPIREProvider(ctx context.Context, cluster *clusterpb.Cluster, spireValues, spireCRDsValues map[string]any, opts ...Option) (*HelmSPIREProvider, error) {
```

Comment:
Should we pass in the kubeconfig unconditionally? I can't think of a case where we wouldn't want to provide a kubeconfig. (we should probably also pass it in when creating a provider for deploy/tear down).